docs: add a section on simple scripts (#96)
geniusyield/atlas-docs: src/pages/additional-features/simple-scripts.mdx

> # Simple Scripts

Reviewer: ```suggestion
# Simple Scripts in Atlas: Native and Timelock Scripts in Cardano
```

Author: I'd just prefer "Simple Scripts", why ".. in Atlas"? And Native/Timelock are just other names of simple scripts.

> # Simple Scripts

Reviewer: ```suggestion
# Simple Scripts in Atlas: Native and Timelock Scripts in Cardano
```

> Atlas supports simple scripts, aka native and timelock scripts. The functionality of simple scripts is explained in ledger design document of Shelley & Mary era [here](https://github.com/IntersectMBO/cardano-ledger) and also in [cardano-node-wiki](https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/reference/simple-scripts.md).

Reviewer: ```suggestion
Atlas supports simple scripts, also known as native and timelock scripts. The functionality of simple scripts is explained in the ledger design documents from the Shelley and Mary eras, which you can find [here](https://github.com/IntersectMBO/cardano-ledger) and in the [cardano-node wiki](https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/reference/simple-scripts.md).
```

> Atlas supports simple scripts, also known as native and timelock scripts. The functionality of simple scripts is explained in the ledger design document of Shelley & Mary era [here](https://github.com/IntersectMBO/cardano-ledger) and also in [cardano-node-wiki](https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/reference/simple-scripts.md).

Reviewer: ```suggestion
Atlas provides support for simple scripts, also referred to as native and timelock scripts, which are key components in building transactions on the Cardano blockchain. These scripts are critical for establishing spending conditions, such as multisig requirements or timelocks. The functionality of simple scripts is detailed in the ledger design document for the Shelley and Mary eras, which can be found [in the Cardano Ledger repository](https://github.com/IntersectMBO/cardano-ledger). Additional documentation can be accessed in the [Cardano Node Wiki's reference section on simple scripts](https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/reference/simple-scripts.md).
```

> Simple scripts can be specified via exposed constructors of [`GYSimpleScript`](https://haddock.atlas-app.io/GeniusYield-Types-Script-SimpleScript.html#t:GYSimpleScript). They can also be read from their JSON representation as `GYSimpleScript` has a [`FromJSON`](https://hackage.haskell.org/package/aeson-2.2.3.0/docs/Data-Aeson.html#t:FromJSON) instance, we also provide a utility function, [`readSimpleScript`](https://haddock.atlas-app.io/GeniusYield-Types-Script-SimpleScript.html#v:readSimpleScript), to read JSON representation from a file.

Reviewer: ```suggestion
Simple scripts can be specified using the exposed constructors of [`GYSimpleScript`](https://haddock.atlas-app.io/GeniusYield-Types-Script-SimpleScript.html#t:GYSimpleScript). They can also be loaded from their JSON representation, as `GYSimpleScript` implements a [`FromJSON`](https://hackage.haskell.org/package/aeson-2.2.3.0/docs/Data-Aeson.html#t:FromJSON) instance. Additionally, we provide the utility function [`readSimpleScript`](https://haddock.atlas-app.io/GeniusYield-Types-Script-SimpleScript.html#v:readSimpleScript), which reads the JSON representation from a file.
```

> Simple scripts can be specified via exposed constructors of [`GYSimpleScript`](https://haddock.atlas-app.io/GeniusYield-Types-Script-SimpleScript.html#t:GYSimpleScript). They can also be read from their JSON representation as `GYSimpleScript` has a [`FromJSON`](https://hackage.haskell.org/package/aeson-2.2.3.0/docs/Data-Aeson.html#t:FromJSON) instance. We also provide a utility function, [`readSimpleScript`](https://haddock.atlas-app.io/GeniusYield-Types-Script-SimpleScript.html#v:readSimpleScript), to read JSON representation from a file.

Reviewer: ```suggestion
Simple scripts can be specified via exposed constructors of the `GYSimpleScript` data type, which represents native and timelock scripts in Atlas. This data type can be used for both creating and interacting with simple scripts programmatically. 

You can specify simple scripts in their JSON representation, which is useful for reading and writing scripts from external files. The `GYSimpleScript` data type implements the [`FromJSON`](https://hackage.haskell.org/package/aeson-2.2.3.0/docs/Data-Aeson.html#t:FromJSON) typeclass, allowing for easy conversion from JSON. Additionally, we provide the [`readSimpleScript`](https://haddock.atlas-app.io/GeniusYield-Types-Script-SimpleScript.html#v:readSimpleScript) utility function to load a script from a JSON file, simplifying the process of working with pre-defined scripts.
```

Author: This suggestion, "...type, which represents native and timelock scripts in Atlas. This data type can be used for both creating and interacting with simple scripts programmatically. " is unnecessary.

> Simple scripts can also be attached to UTxO outputs to be used as reference script inputs later when building transaction. We have an example [here](https://github.com/geniusyield/atlas/blob/main/tests-privnet/GeniusYield/Test/Privnet/SimpleScripts.hs), demonstrating interaction with simple scripts in both scenarios, when being used as reference and otherwise.

Reviewer: ```suggestion
Simple scripts can also be attached to UTxO outputs, allowing them to be used later as reference scripts when building transactions. You can find an example [here](https://github.com/geniusyield/atlas/blob/main/tests-privnet/GeniusYield/Test/Privnet/SimpleScripts.hs), that demonstrates how to interact with simple scripts, both as reference scripts and in other scenarios.
```

> Simple scripts can also be attached to UTxO outputs to be used as reference script inputs later when building transactions. We have an example [here](https://github.com/geniusyield/atlas/blob/main/tests-privnet/GeniusYield/Test/Privnet/SimpleScripts.hs), demonstrating interaction with simple scripts in both scenarios, when being used as reference and otherwise.

Reviewer: ```suggestion
Simple scripts can also be attached to UTxO outputs, enabling them to be referenced later when building transactions. This allows for reusability of the scripts without embedding them directly into every transaction. We provide an example [here](https://github.com/geniusyield/atlas/blob/main/tests-privnet/GeniusYield/Test/Privnet/SimpleScripts.hs), demonstrating the interaction with simple scripts in two scenarios: when used as reference scripts and when they are embedded directly.
```

> Simple scripts can also be attached to UTxO outputs to be used as reference script inputs later when building transactions. We have an example [here](https://github.com/geniusyield/atlas/blob/main/tests-privnet/GeniusYield/Test/Privnet/SimpleScripts.hs), demonstrating interaction with simple scripts in both scenarios, when being used as reference and otherwise.

Reviewer: ```suggestion
Additionally, we provide the [`readSimpleScript`](https://haddock.atlas-app.io/GeniusYield-Types-Script-SimpleScript.html#v:readSimpleScript) utility function to load a script from a JSON file, simplifying the process of working with pre-defined scripts.
```
